util/blockdevice/util_test.go

```diff
@@ -124,13 +231,554 @@ func TestGetCapacityOrError(t *testing.T) {
 		name := name
 		mock := mock
 		t.Run(name, func(t *testing.T) {
-			_, err := GetCapacityOrError(mock.src)
+			_, err := GetPhysicalSectorSizeOrError(mock.src)
+			if mock.isErr && err == nil {
+				t.Fatalf("Expected error got none")
+			}
+			if !mock.isErr && err != nil {
+				t.Fatalf("Expected no error got [%+v]", err)
+			}
+		})
+	}
+}
+
+func TestGetHostNameOrError(t *testing.T) {
+	var tests = map[string]struct {
+		src    unstructured.Unstructured
+		result string
+		isErr  bool
+	}{
+		"nil object": {
+			src: unstructured.Unstructured{
+				Object: nil,
+			},
+			isErr: true,
+		},
+		"empty object": {
+			src:   unstructured.Unstructured{},
+			isErr: true,
+		},
+		"kind mismatch error": {
+			src: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind": "test",
+				},
+			},
+			isErr: true,
+		},
+		"not found error": {
+			src: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind": string(types.KindBlockDevice),
+				},
+			},
+			isErr: true,
+		},
+		"empty value error": {
+			src: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind": string(types.KindBlockDevice),
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"kubernetes.io/hostname": "",
+						},
+					},
+				},
+			},
+			isErr: true,
+		},
+		"type mismatch error": {
+			src: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind": string(types.KindBlockDevice),
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"kubernetes.io/hostname": 1,
+						},
+					},
+				},
+			},
+			isErr: true,
+		},
+		"valid hostname label": {
+			src: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind": string(types.KindBlockDevice),
+					"metadata": map[string]interface{}{
+						"labels": map[string]interface{}{
+							"kubernetes.io/hostname": "my-host-1",
+						},
+					},
+				},
+			},
+			isErr:  false,
+			result: "my-host-1",
+		},
+	}
+	for name, mock := range tests {
+		name := name
+		mock := mock
+		t.Run(name, func(t *testing.T) {
+			result, err := GetHostNameOrError(mock.src)
+			if mock.isErr && err == nil {
+				t.Fatalf("Expected error got none")
+			}
+			if !mock.isErr && err != nil {
+				t.Fatalf("Expected no error got [%+v]", err)
+			}
+			if !mock.isErr && result != mock.result {
+				t.Fatalf("Expected host name %q got %q", mock.result, result)
+			}
+		})
+	}
+}
+
+func TestGetNodeNameOrError(t *testing.T) {
+	var tests = map[string]struct {
+		src    unstructured.Unstructured
+		result string
```

Comment:
can result be renamed to expectNodeName or expect

Reply:
made it expectedNodeName